File: C++/48-不用加减乘除做加法.cpp
class Solution
{
public:
    int Add(int num1, int num2)
    {
        int MIN = 0xffffffff;
        int mask = 0xffffffff;
        while (num2){
            int t = (num1 ^ num2) & mask;
            num2 = (num1 & num2) << 1 & mask;
            num1 = t;
        }
        return num1;
    }
};

File: Python/48-不用加减乘除做加法.py
# -*- coding:utf-8 -*-
class Solution:
    def Add(self, num1, num2):
        # write code here
        mask = 0xffffffff
        while num2:
            t = (num1 ^ num2) & mask
            num2 = (num1 & num2) << 1 & mask
            num1 = t
        if num1 <= 0x7fffffff:
            return num1
        else:
            return ~(num1 ^ mask)